packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -192,11 +193,25 @@ const scriptReplacer = (
 export type BootstrapScriptDescriptor = {
   src: string,
   integrity?: string,
+  crossOrigin?: string,
 };
 export type ExternalRuntimeScript = {
   src: string,
   chunks: Array<Chunk | PrecomputedChunk>,
 };
+
+/**
+ * We normalize crossOrigin to be "use-credentials", "" or null.
+ * Unless using "use-credentials" all the other strings behave like "anonymous" and "" is
+ * shortest bytes-wise.
+ */
+function getCrossOrigin(options?: mixed) {
```

Comment:
I'd revert the commit where you extract this function and leave it inline.

It works today that the script desciptors and the preconnect options happen to correlate and so this function works but it's essentially duck-typing a mixed object that provides no guarantees it will continue to stay in the shape it currently is in.

We'd hope closure would also inline these functions anyway and the logic isn't particularly complicated so we should just leave them inline

A compromise might be to only extract the function

```
function getCrossOriginFromString(crossOriginString: string): string {
  return crossOriginString === 'use-credentials' ? crossOriginString : ""
}
```

and then just use it once you've checked that the cross origin config is known to not be null. closure will almost certainly inline this and it does not presume any particular shape of options/configuration

Reply:
Sure! Let me do this.